clisops/core/regrid.py

```diff
@@ -0,0 +1,1861 @@
+    format: str, optional
+        Not yet implemented. When reading weights from disk, the input format may be specified.
+        If omitted, there will be an attempt to detect the format. The default is None.
+    """
+
+    @require_xesmf
+    def __init__(
+        self,
+        grid_in: Grid,
+        grid_out: Grid,
+        method: str,
+        from_disk: str | Path | None = None,
+        format: str | None = None,
+    ):
+        """Initialize Weights object, incl. calculating / reading the weights."""
+        if not isinstance(grid_in, Grid) or not isinstance(grid_out, Grid):
+            raise InvalidParameterValue(
+                "Input and output grids have to be instances of clisops.core.Grid."
+            )
+        self.grid_in = grid_in
+        self.grid_out = grid_out
+        self.method = method
+
+        # Compare source and target grid
+        if grid_in.hash == grid_out.hash:
+            raise Exception(
+                "The selected source and target grids are the same. "
+                "No regridding operation required."
+            )
+
+        # Periodic in longitude
+        # TODO: properly test / check the periodic attribute of the xESMF Regridder.
+        #  The grid.extent check done here might not be suitable to set the periodic attribute:
+        #  global == is grid periodic in longitude
+        self.periodic = False
+        try:
+            if self.grid_in.extent == "global":
+                self.periodic = True
+        except AttributeError:
+            # forced to False for conservative regridding in xesmf
+            # TODO: check if this is proper behaviour of xesmf
+            if self.method not in ["conservative", "conservative_normed"]:
+                warnings.warn(
+                    "The grid extent could not be accessed. "
+                    "It will be assumed that the input grid is not periodic in longitude."
+                )
+
+        # Activate ignore degenerate cells setting if collapsing cells are found within the grids.
+        #  The default setting within ESMF is None, not False!
+        self.ignore_degenerate = (
+            True
+            if (
+                self.grid_in.contains_collapsed_cells
+                or self.grid_out.contains_collapsed_cells
+            )
+            else None
+        )
+
+        self.id = self._generate_id()
+        self.filename = "weights_" + self.id + ".nc"
+
+        if not from_disk:
+            # Read weights from cache or compute & save to cache
+            self.format = "xESMF"
+            self._compute()
+        else:
+            # Read weights from disk
+            self._load_from_disk(filename=from_disk, format=format)
+
+        # Reformat and cache the weights if required
+        # FIXME: _detect_format() and _reformat() are not implemented yet
+        if not self.tool.startswith("xESMF"):
+            self.format = self._detect_format()
+            self._reformat("xESMF")
```

Comment:
This should be commented or raise appropriate warnings/errors

Reply:
I added an error message